File: python/Problem010.py
#
# This file contains a solution to the problem number 10 of the Project Euler, last modified on 5/13/2018.
# Copyright (c) Matthieu Le Boucher. All rights reserved.
#

import EulerUtils

def result():
    n = 2_000_000
    sieve = EulerUtils.sieve(n)

    sum = 2
    for i in range(1, int((n - 1) / 2)):
        if not sieve[i]:
            sum += 2 * i + 1

    return sum

if __name__ == "__main__":
    print(result())

File: cpp/Problem010.cpp
#include "EulerSolution.h"
#include <cmath>
#include "EulerUtils.h"
#include <vector>

/*
 * @author Matthieu Le Boucher <matt.leboucher@gmail.com>
 */

using namespace std;

class Problem010 : public EulerSolution {
public:
    int number() override {
        return 10;
    }

    double solution() override {
        int limit = 2000000;
        vector<bool> sieve = Arithmetics.getSieve(limit);

        double sum = 2;
        for (int i = 1; i <= (limit - 1) / 2; i++)
            if (!sieve[i]) sum += 2 * i + 1;

        return sum;
    }

    double answer() override {
        return 31875000;
    }
};